lec15-git-projects.Rmd

```diff
@@ -0,0 +1,415 @@
+
+Material covered in class, in order:
+
+- [Creating a repository](http://swcarpentry.github.io/git-novice/03-create/index.html)
+- [Tracking changes](http://swcarpentry.github.io/git-novice/04-changes/index.html)
+- [Exploring history](http://swcarpentry.github.io/git-novice/05-history/index.html)
+
+## Collaborating with GitHub
+
+Although command line Git is very useful, GitHub allows for easy use of
+Git for collaborative purposes using a primarily point-and-click interface,
+in addition to providing a web-based hosting service for Git repositories (or 'repos'). 
+If you have not already made a GitHub account, do so now [here](www.github.com).
+
+All groups have been provided with existing repositories, but a new
+repository can be made by clicking on the `+` in the top right of the page
+and selecting 'New repository'. For now, however, we'll navigate to
+our provided group repos.
+
+### Creating a fork
+
+The repos that have already been created can be thought of as 'master
+repos', which will contain the 'primary' version of the repo at any
+given point in time. However, instead of directly uploading and editing
+files right within this main repo itself, we will begin by _forking_ the
+repo. When a given user forks a repo, GitHub creates a user-specific copy
+of the repo and all its files.
+
+To fork your group's repo, navigate to the repo's page and click on the
+'Fork' button at the top right of the page. Following a brief load
+screen, GitHub will redirect you to your new, forked repo.
+
+You'll notice that on the top left of this repo page, the repo's name
+will be '[your username] / [repo name]', as opposed to 'EEB313 / [repo name]'.
+Furthermore, GitHub will indicate that this is a fork right underneath
+said repo name ('forked from eeb313-2018/ [repo name]'). 
+
+### Branches and Pull Requests
+
+Currently, your fork is identical to what the original repo looked like
+at the moment of forking, and so GitHub will indicate that the fork
+is currently 'even' with its original. Any edits that are made in
+the fork will be unique to the fork, leaving the original unchanged.
+
+But what if we do want our changes to show on the main repo? This is
+a two step process:
+
+1. First, we create a _branch_ on our _fork_ (not the main repo). This is yet 
+another copy of the fork (a copy of a copy!) that will store changes separately 
+from 'master', which is the default branch. 
+
+2. Once we have made our changes in this branch, we will submit what's
+known as a _pull request_ from this edited branch to the main repo. This 
+allows your changes to be reviewed by the other members of your group before 
+they are merged into the main repo.
+
+Although this process may seem a bit laborious, using this method (also known as
+the 'GitHub flow') minimizes chances of error and ensures that all code
+is reviewed by more than one person. Understanding how and why this process works
```

Comment:
At least one person? Unless you are counting the original author as also reviewing code.